delfin/drivers/dell_emc/unity/unity.py

```diff
@@ -0,0 +1,204 @@
+                        pool.get('content').get('id')),
+                    'description': pool.get('content').get('description'),
+                    'status': status,
+                    'storage_type': pool_type,
+                    'total_capacity': int(pool.get('content').
+                                          get('sizeTotal')),
+                    'subscribed_capacity': int(pool.get('content').get(
+                        'sizeSubscribed')),
+                    'used_capacity': int(pool.get('content').get('sizeUsed')),
+                    'free_capacity': int(pool.get('content').get('sizeFree'))
+                }
+                pool_list.append(p)
+        return pool_list
+
+    def volume_handler(self, volumes, volume_list):
+        if volumes is not None:
+            vol_entries = volumes.get('entries')
+            for volume in vol_entries:
+                total = volume.get('content').get('sizeTotal')
+                used = volume.get('content').get('sizeAllocated')
+                vol_type = constants.VolumeType.THICK
+                if volume.get('content').get('isThinEnabled') is True:
+                    vol_type = constants.VolumeType.THIN
+                compressed = True
+                deduplicated = volume.get('content').\
+                    get('isAdvancedDedupEnabled')
+                health_value = volume.get('content').get('health').get('value')
+                if health_value == 5:
+                    status = constants.StorageStatus.NORMAL
+                else:
+                    status = constants.StorageStatus.ABNORMAL
+                v = {
+                    'name': volume.get('content').get('name'),
+                    'storage_id': self.storage_id,
+                    'description': volume.get('content').get('description'),
+                    'status': status,
+                    'native_volume_id': str(volume.get('content').get('id')),
+                    'native_storage_pool_id':
+                        volume.get('content').get('pool').get('id'),
+                    'wwn': volume.get('content').get('wwn'),
+                    'type': vol_type,
+                    'total_capacity': int(total),
+                    'used_capacity': int(used),
+                    'free_capacity': int(total - used),
+                    'compressed': compressed,
+                    'deduplicated': deduplicated
+                }
+                volume_list.append(v)
+
+    def list_volumes(self, context):
+        page_size = 1
+        volume_list = []
+        while True:
+            luns = self.rest_handler.get_all_luns(page_size)
+            if 'entries' not in luns:
+                break
+            if len(luns['entries']) < 1:
+                break
+            self.volume_handler(luns, volume_list)
+            page_size = page_size + 1
+        page_size = 1
+        while True:
+            filesystems = self.rest_handler.get_all_filesystem(page_size)
```

Comment:
Should we consider filesystems as volumes? this is a NAS capability , In my view , if a filesystem has a backing LUN we should consider it as volume, otherwise including filesystems in volumes will create confusion. Unity also has VVoLs , I don't see a fucntion to list vvols ..

Reply:
i have checked this,you are right,i'll change that

Reply:
done